File: leetcode/Q160_IntersectionofTwoLinkedLists.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> ListNode:
        nodes = {}
        temp = headA
        while temp is not None:
            nodes.add(temp)
            temp = temp.next

        temp = headB
        while temp is not None:
            if temp in nodes:
                return temp
            temp = temp.next

        return None

File: src/easy/Q160_IntersectionofTwoLinkedLists.java
package easy;

import java.util.HashSet;

public class Q160_IntersectionofTwoLinkedLists {
    class ListNode {
        int val;
        ListNode next;
        ListNode(int x) {
          val = x;
          next = null;
        }
    }
    public class Solution {
        public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
            HashSet<ListNode> nodeHashSet = new HashSet<>();
            while (headA != null){
                nodeHashSet.add(headA);
                headA = headA.next;
            }
            while (headB != null){
                if(nodeHashSet.contains(headB)){
                    return headB;
                }
                headB = headB.next;
            }
            return null;
        }
    }
}
